lib/grid.rb

```diff
@@ -50,4 +62,34 @@ def coordinates_in_grid?(coordinates)
   def coordinates_unmarked?(coordinates)
     available_coordinates.include?(coordinates)
   end
+
+  def computer_coordinates
+    cells.select { |_, value| value == "O" }.keys.to_set
+  end
+
+  def player_coordinates
+    cells.select { |_, value| value == "X" }.keys.to_set
+  end
+
+  def winner?(cells)
+    winning_combinations.any? { |combination|
+      combination.subset?(cells)
+    }
+  end
+
+  def winning_combinations
+    rows + columns + diagonals
+  end
+
+  def rows
+    [Set["A1", "A2", "A3"], Set["B1", "B2", "B3"], Set["C1", "C2", "C3"]]
```

Comment:
Used sets as to not be concerned about the order. Just looking at these methods feels wrong, though I can't really communicate why.

Reply:
I'm not sure the methods themselves feel wrong. I think the whole determination of winning is not a responsibility of the grid. I think the grid can return data that a `GameStatus` class could use to do this stuff. 

Although, the angle brackets as a creator feels a bit confusing. Legit Ruby, but not an expected way to initialize an object. I like the use of set.